presto-main/src/main/java/com/facebook/presto/execution/DedupingWarningCollector.java

```diff
@@ -14,25 +14,33 @@
 package com.facebook.presto.execution;
 
 import com.facebook.presto.spi.PrestoWarning;
+import com.facebook.presto.spi.QueryId;
 import com.google.common.collect.ImmutableList;
 
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
 public class DedupingWarningCollector
         implements WarningCollector
 {
     private final Set<PrestoWarning> warnings = new LinkedHashSet<>();
+    private final QueryId queryId;
 
-    public DedupingWarningCollector()
+    public DedupingWarningCollector(QueryId queryId)
     {
+        this.queryId = requireNonNull(queryId, "queryId is null");
     }
 
     @Override
     public void add(PrestoWarning warning)
     {
-        warnings.add(warning);
+        warnings.add(new PrestoWarning(
+                warning.getCode(),
+                format("%s: %s", queryId, warning.getMessage())));
```

Comment:
Why do we need to prefix the query ID? I see that the `QueryStateMachine` gets these warnings and stick them into `QueryInfo` instance, which already knows the query ID, which makes me think this is redundant.

Reply:
Agreed, we should not add query ID here.